File: hotel-master-reservation-client/src/app/model/common/base.service.ts
import { tap } from 'rxjs/operators';
import { Subject } from 'rxjs';
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';

@Injectable()
export abstract class BaseService<T>{

    protected abstract url(): string;
    dataChanged = new Subject<T>();

    constructor(protected http: HttpClient){}

    save(t: T){
        return this.http.post<T>(this.url(), t);
    }

    findAll(){
        return this.http.get<T[]>(this.url());
    }

    findById(id){
        return this.http.get<T>(`${this.url()}/${id}`).pipe(
            tap(hotel => this.dataChanged.next(hotel))
        )
    }
}

File: hotel-core-domain/src/main/java/com/solt/hotel/master/common/BaseService.java
package com.solt.hotel.master.common;

import java.util.List;

public interface BaseService<T, ID> {

	BaseRepository<T, ID> repo();
	
	default T save(T t) {
		return repo().save(t);
	}
	
	default List<T> findAll(){
		return repo().findAll();
	}

	default T findById(ID id) {
		return repo().getOne(id);
	}
}
